main.go

```diff
@@ -130,6 +135,13 @@ func initServer() *http.Server {
 	return &http.Server{Addr: address, Handler: router}
 }
 
+func runServer(server *http.Server) error {
+	if tlsCertFile != "" && tlsKeyFile != "" {
```

Comment:
nit: I'm wondering whether we need to introduce some basic validation here (to check if the file exist or similar). Otherwise we may end up with a generic HTTP server error.